### PR TITLE
implement `GroupNorm2D` from paper

### PR DESCRIPTION
**Pull request by davexpro**
Paper: [Group Normalization](https://arxiv.org/abs/1803.08494)

Migrated From TF code given in paper:

```
def GroupNorm(x, gamma=1, beta=0.5, group=32, eps=1e-5):
    # x: input features with shape [N,C,H,W]
    # gamma, beta: scale and offset, with shape [1,C,1,1]
    # G: number of groups for GN
    
    N, C, H, W = x.shape
    x = tf.reshape(x, [N, group, C // group, H, W])
    
    mean, var = tf.nn.moments(x, [2, 3, 4], keep_dims=True)
    x = (x - mean) / tf.sqrt(var + eps)
    
    x = tf.reshape(x, [N, C, H, W])
    
    return x * gamma + beta
```

I test it with VGG16 and cifar10, it works well.

### CODE REVIEW

**Comment by rasbt** [comment]
Looks nice, but why did you put the tensorflow code in the docstring? If someone's running doctests, that will cause problems

**Comment by DavexPro** [comment]
well, I don't realize that...How to fix that?

**Comment by rasbt** [comment]
I would suggest removing it, because it's already referenced in the paper, which is mentioned in the docstring. Alternatively, this could be mentioned in the documentation, but I think it doesn't add to the clarity because it's almost identical to the PyTorch code :)

**Comment by rasbt** [comment]
Ah nice, I see that the tests pass now :)

**Comment by DavexPro** [comment]
Terrific! What should I do next?

**Comment by rasbt** [comment]
I think someone of the core devs needs to take a look at it (I just stumbled upon this PR since I wanted to upload mine almost at the same time after I read the He paper this morning :P)

**Comment by soumith** [comment]
thanks for the PR, we have a WIP implementation of more efficient code that implements GroupNorm (dedicated GPU code) that @SsnL was going to PR today or tomorrow (as soon as the paper was out :) ). Additional to the implementation, one needs tests, which @ssnl will add.

I think adding a torch.nn.moments is also a good idea, seems pretty useful.

**Comment by DavexPro** [comment]
I found my code is incomplete, after asking Wu, gamma & beta is a trainable, this code is just like the basic func, looking forward to the efficient one =)
